nexus/src/app/sagas/instance_create.rs

```diff
@@ -626,13 +626,15 @@ async fn sic_allocate_instance_snat_ip(
         OpContext::for_saga_action(&sagactx, &saga_params.serialized_authn);
     let instance_id = sagactx.lookup::<Uuid>("instance_id")?;
     let ip_id = sagactx.lookup::<Uuid>("snat_ip_id")?;
+
+    let (.., pool) = datastore
+        .ip_pools_lookup_by_name_no_auth(&opctx, "default")
```

Comment:
Is this really just for looking up the `default` IP pool? If so, I wonder if we shouldn't just have a `ip_pools_lookup_default` that's documented to not require authz but otherwise doesn't take arbitrary names.

Reply:
I did this, and added authz checks, within 91415a4571c916c88c9432fc5ad7620722191afe